File: hw/medina/u1/HW11ChickenCoops/newChickenCoopPython/model/ChickenCoop.py
class ChickenCoop:
    def __init__(self, id: int, name: str):
        self.id = id
        self.name = name
        self.chickens = []  

    def add_chicken(self, chicken):
        self.chickens.append(chicken)

    def get_chickens(self):
        return self.chickens

    def __str__(self):
        chickens_info = "\n".join([str(ch) for ch in self.chickens])
        return f"\nChickenCoop{{ id -> {self.id}, name -> {self.name}, chickens -> [{chickens_info}] }}"

File: hw/medina/u1/HW11ChickenCoops/newChickenCoopJS/model/ChickenCoop.js
/**
 * ChickenCoop class
 * Represents a coop containing multiple chickens.
 */
class ChickenCoop {
  constructor(id, name) {
    this.id = id;
    this.name = name;
    this.chickens = []; 
  }

  addChicken(chicken) {
    this.chickens.push(chicken);
  }

  getChickens() {
    return this.chickens;
  }

  toString() {
    const chickensInfo = this.chickens.map(ch => ch.toString()).join("\n");
    return `\nChickenCoop{ id -> ${this.id}, name -> ${this.name}, chickens -> [${chickensInfo}] }`;
  }
}

module.exports = { ChickenCoop };
